lib/main/server.js

```diff
@@ -19,39 +20,34 @@ Server.prototype.processRequest = function(request, response) {
   var urlParams = url.parse(request.url, true);
   if (request.method === 'GET' && urlParams.pathname === '/I/want/title/') {
     var addresses = (typeof(urlParams.query.address) === 'string') ? [urlParams.query.address] : urlParams.query.address;
-    var titles = [];
     var body = [];
 
-    function processTitle(title, index){
-      titles.push(title);
-      if(index < addresses.length - 1){
-          console.log(this);
-          self.fetchTitle(addresses[index + 1], index + 1, processTitle);
-      } else {
+    async.map(addresses, self.fetchTitle, function(error, result){
```

Comment:
A small thing but name the variable as 'result' might be misleading as it is an array.

Rest is beautiful.